expression/builtin_op.go

```diff
@@ -678,25 +678,7 @@ func (c *isNullFunctionClass) getFunction(ctx context.Context, args []Expression
 	if err := c.verifyArgs(args); err != nil {
 		return nil, errors.Trace(err)
 	}
-	tc := args[0].GetType().ToClass()
-	var argTp evalTp
-	switch tc {
-	case types.ClassInt:
-		argTp = tpInt
-	case types.ClassDecimal:
-		argTp = tpDecimal
-	case types.ClassReal:
-		argTp = tpReal
-	default:
-		tp := args[0].GetType().Tp
-		if types.IsTypeTime(tp) {
-			argTp = tpDatetime
-		} else if tp == mysql.TypeDuration {
-			argTp = tpDuration
-		} else {
-			argTp = tpString
-		}
-	}
+	argTp := fieldTp2EvalTp(args[0].GetType())
```

Comment:
We should check tpDatetime and tpTimestamp here,
if tpDatetime || tpTimestamp {
    argTp = tpDatetime
}